### ISSUE TITLE
rustup show: Can, instead, configure your system (silently)

### ISSUE THREAD

**Issue by bbros-dev** [rust-lang/rustup#2957]
### Problem

When `~/.rustup` does not exist and `RUSTUP_HOME` and `CARGO_HOME` are not set.  Running `rustup show`, ***silently*** creates the folder `~/.rustup` and the file `~/.rustup/settings.toml`.

This

### Steps

1.
```
~$ ls ~/.rustup
ls: cannot access '/home/<user>/.rustup': No such file or directory
```

2.
```
~$ echo $RUSTUP_HOME 
/home/<user>/.local/share/rustup
```
3.
```
~$ echo $CARGO_HOME 
/home/<user>/.local/share/cargo
```
4.
```
~$ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/<user>/.rustup
```
5.
```
~$ ls ~/.rustup
settings.toml
```

### Possible Solution(s)

```cucumber
Given the folder `~/.rustup` does not exist
  And the environment variable RUSTUP_HOME is not set
  And the environment variable CARGO_HOME is not set
 When a user runs `rustup show`
 Then the folder `~/.rustup` is not created
  And the stderr output does contain:
"""
error: rustup is not installed at '/home/<snip>/.rustup' and RUSTUP_HOME is not set.  One is required.
"""
```

Another foreseeable scenario

```cucumber
Given the folder `~/.rustup` does not exist
  And the environment variable RUSTUP_HOME is set
  And the environment variable CARGO_HOME is not set
  And the folder pointed to by RUSTUP_HOME does not exist
 When a user runs `rustup show`
 Then the folder `~/.rustup` is not created
  And the folder pointed to by RUSTUP_HOME is not created
  And the stderr output does contain:
"""
error: RUSTUP_HOME points to the folder /some/path, that does not exist. The file RUSTUP_HOME/settngs.toml is also required. See http://some/doc/url.html
"""
```

### Notes

This was [observed](https://github.com/rust-lang/rustup/issues/2886#issuecomment-1022664072) in the course of issue #2886 

### Rustup version

```console
$ rustup --version
rustup 1.24.3 (ce5817a94 2021-05-31)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.59.0 (9d1b2106e 2022-02-23)`
```

### Installed toolchains

```console
$ rustup show                                                                               
Default host: x86_64-unknown-linux-gnu                                                            
rustup home:  /home/hedge/.local/share/rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
nightly-2021-09-01-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
1.45.2-x86_64-unknown-linux-gnu
1.52.1-x86_64-unknown-linux-gnu
1.53.0-x86_64-unknown-linux-gnu
1.54.0-x86_64-unknown-linux-gnu
1.56.1-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.59.0 (9d1b2106e 2022-02-23)
```

**Comment by jyn514** [comment]
Why is this a bug? Rustup needs to manage these directories, it's an implementation detail when they're created.

**Comment by bbros-dev** [comment]
> Why is this a bug?

To my mind:  because it is not `show`ing it is initializing.  

I referred to the issue where this was uncovered.  
Under the conditions set out I'd expect something like what was set out above - the error text above could be improved, no argument here. In fact I've added another scenario to the OP. 

If the behavior was as set out above, `rustup show` would have indicated there is an issue with how the env var `RUSTUP_HOME` is being set, because `rustup ...` doesn't have access to it.

I'm strongly opposed to a command called `show` making ***any*** changes to my system - I don't believe that is a reasonable thing to do to users.

As a tactical workaround pending a fix it would be courteous to print a warning that `rustup show` is about to change the configuration of your system, maybe something like:

```
Warning: 
Unable to find RUSTUP_HOME, nor the folder ~/.rustup nor the file ~/.rustup/settings.toml.  
We know you asked to show the current configuration details.  
Instead we've decided to take it upon ourselves to configure your system for you by creating that folder and file.
```

Hopefully that makes the objection clear.
